### PR TITLE
Move around pool setup logic to re-importable infra

### PR DESCRIPTION
**Pull request by ValarDragon**
The purpose of this PR is to add in some scaffolding for testing the spot price. Adam mentioned that the part he wasn't sure how to do for #1155 / general osmosis code development was the testing infrastructure. So the goal of this PR is to:

- Update more of the testing code to use the `apptesting` module
- Make a table driven test in balancer that can be adapted for working on #1155 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by ValarDragon** [comment]
@czarcas7ic check out the second commit ( https://github.com/osmosis-labs/osmosis/pull/1156/commits/acbdb8668eb76132eda7a9e33229bd3a6043e7ab )

Basically the way you'd test #1155 with this change is then by editing / adding more test cases in that test, as you do the change. You can setup your editor to run those tests every time you save a file in the balancer package.

File: x/gamm/pool-models/balancer/suite_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+
+	"github.com/osmosis-labs/osmosis/v7/app"
+	"github.com/osmosis-labs/osmosis/v7/app/apptesting"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
```
Comment (alexanderbez) [inline x/gamm/pool-models/balancer/suite_test.go:14]: formatting is off here 👍

**Review by alexanderbez — APPROVED** [review]
LGTM!

Question, `SetupTest()` is called by Testify's suite before _each_  test. Why do we also call it for each test case?

**Comment by ValarDragon** [comment]
Re SetupTest, maybe we're doing it wrong!

The idea right now is we want it to start on a clean app state, and we set the clean app state on SetupTest. Should this per test-case reset state method have a different name?

**Comment by alexanderbez** [comment]
> Re SetupTest, maybe we're doing it in a way thats against the standard!
> 
> The idea right now is we want it to start on a clean app state, and we set the clean app state on SetupTest. Should this per test-case reset state method have a different name?

Nope, that's totally fine and I assumed as much!

**Pull request merged**